### PR TITLE
Add support for OpenShift distribution

### PR DESCRIPTION
**Pull request by owais**

### CODE REVIEW

File: helm-charts/splunk-otel-collector/templates/clusterRole.yaml [hunk 1]
```diff
@@ -87,3 +97,48 @@ rules:
   - get
   - list
   - watch
+---
+{{- if eq .Values.distro "openshift" }}
+kind: SecurityContextConstraints
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/clusterRole.yaml:102]: can we move it to a separate yaml? all other resources defined separately

File: helm-charts/splunk-otel-collector/templates/clusterRole.yaml [hunk 2]
```diff
@@ -87,3 +97,48 @@ rules:
...
+defaultAddCapabilities: []
+fsGroup:
+  type: MustRunAs
+readOnlyRootFilesystem: false
+runAsUser:
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+requiredDropCapabilities: []
+{{- end }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/clusterRole.yaml:144]: nit: no empty line at the end

File: helm-charts/splunk-otel-collector/templates/clusterRole.yaml [hunk 3]
```diff
@@ -87,3 +97,48 @@ rules:
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+users:
+- system:serviceaccount:{{ .Release.Namespace }}:{{ template "splunk-otel-collector.serviceAccountName" . }}
+priority: 10
+allowHostNetwork: true 
+allowHostPorts: true
+allowHostPID: true
+allowHostDirVolumePlugin: true
+allowHostIPC: false
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/clusterRole.yaml:118]: Is the access to IPC namespace required? I may be wrong but I think it's not needed for default chart functionality, not sure about openshift

**Comment by dmitryax** [comment]
also please run `make render`

File: helm-charts/splunk-otel-collector/templates/clusterRole.yaml (hunk 3 above)
Comment (owais) [reply helm-charts/splunk-otel-collector/templates/clusterRole.yaml:118]: It is set to `false`. Are you saying it should be true?

**Comment by owais** [comment]
> also please run make render

I did but it didn't generate anything new. Will try again

File: helm-charts/splunk-otel-collector/templates/clusterRole.yaml (hunk 3 above)
Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/clusterRole.yaml:118]: Oh my bad. I misread the boolean value :)

**Review by dmitryax — APPROVED** [review]
LGTM. we can merge it after some more testing

**Pull request merged**